dotcom-rendering/src/components/AudioPlayer/components/WaveForm.tsx

```diff
@@ -97,7 +81,10 @@ export const WaveForm = ({
 	...props
 }: Props) => {
 	// memoise the waveform data so they aren't recalculated on every render
-	const barHeights = useMemo(() => generateWaveform(src, bars), [src, bars]);
+	const barHeights = useMemo(
+		() => Array.from(waveformGenerator(src, bars, [60, 100])),
```

Comment:
i didn't realise you could do this with generators in `Array.from` it's really interesting

Reply:
👀

Reply:
Oh yeah you most definitely can: it’s because they are [Iterators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator), too!

You could technically also `[...waveformGenerator]` but that would require a different target I think.